packages/web/src/common/store/tipping/sagas.ts

```diff
@@ -263,7 +267,171 @@ function* confirmTipIndexed({
   return false
 }
 
+function* wormholeAudioIfNecessary({ amount }: { amount: number }) {
+  const walletClient = yield* getContext('walletClient')
+
+  const waudioBalanceWei = yield* call([
+    walletClient,
+    'getCurrentWAudioBalance'
+  ])
+  const audioWeiAmount = new BN(AUDIO(amount).value.toString()) as BNWei
+
+  if (isNullOrUndefined(waudioBalanceWei)) {
+    throw new Error('Failed to retrieve current wAudio balance')
+  }
+
+  // If transferring spl wrapped audio and there are insufficent funds with only the
+  // user bank balance, transfer all eth AUDIO to spl wrapped audio
+  if (audioWeiAmount.gt(waudioBalanceWei)) {
+    console.info('Converting Ethereum AUDIO to Solana wAUDIO...')
+
+    // Wait for a second before showing the notice that this might take a while
+    const showConvertingMessage = yield* fork(function* () {
+      yield delay(1000)
+      yield put(convert())
+    })
+    yield call([walletClient, walletClient.transferTokensFromEthToSol])
+    // Cancel showing the notice if the conversion was magically super quick
+    yield cancel(showConvertingMessage)
+  }
+}
+
 function* sendTipAsync() {
+  const audiusSdk = yield* getContext('audiusSdk')
+  const sdk = yield* call(audiusSdk)
+  const isNativeMobile = yield* getContext('isNativeMobile')
+
+  const sender = yield* select(getAccountUser)
+  const {
+    trackId,
+    user: receiver,
+    amount: stringAudioAmount,
+    onSuccessConfirmedActions,
+    source
+  } = yield* select(getSendTipData)
+
+  if (!sender || !receiver) {
+    return
+  }
+
+  const device = isNativeMobile ? 'mobile' : 'web'
+
+  const senderUserId = encodeHashId(sender.user_id)
+  const receiverUserId = encodeHashId(receiver.user_id)
+  const amount = Number(stringAudioAmount)
+
+  try {
+    yield put(
+      make(Name.TIP_AUDIO_REQUEST, {
+        senderWallet: sender.userBank,
+        recipientWallet: receiver.userBank,
+        senderHandle: sender.handle,
+        recipientHandle: receiver.handle,
+        amount,
+        device,
+        source
+      })
+    )
+
+    yield* call(wormholeAudioIfNecessary, { amount })
+
+    yield* call([sdk.users, sdk.users.sendTip], {
+      amount,
+      senderUserId,
+      receiverUserId
+    })
+
+    yield put(sendTipSucceeded())
+    yield put(
+      make(Name.TIP_AUDIO_SUCCESS, {
+        recipientHandle: receiver.handle,
+        amount: stringAudioAmount,
+        device,
+        source
+      })
+    )
+
+    yield* put(refreshTipGatedTracks({ userId: receiver.user_id, trackId }))
+
+    yield* fork(function* () {
```

Comment:
Just confirming intention here is to fork of a non-blocking generator and let the code continue on line 389 regardless of when this finishes? Seems like the rest is an optimistic update, so it makes sense if everything below is safe to proceed in parallel.
If I understand correctly, it wouldn't be caught by the catch below if it fails? Or does it wait for the forked generator and the current one to finish and then processes any exceptions after they merge?

Wondering if it would make sense for this to be pulled out into a separate function that can just be called and forgotten about?